Source/CombatExtended/Harmony/Harmony_PawnRenderer.cs

```diff
@@ -31,22 +31,31 @@ private enum WriteState
             WriteShell,
             WritePostShell
         }
-
+        
         // Sync these with vanilla PawnRenderer constants
         private const float YOffsetBehind = 0.00306122447f;
         private const float YOffsetHead = 0.0244897958f;
         private const float YOffsetOnHead = 0.0306122452f;
         private const float YOffsetPostHead = 0.03367347f;
         private const float YOffsetIntervalClothes = 0.00306122447f;
 
+        private static Rot4 north = Rot4.North; // this otherwise creates a new instance per invocation
+
         private static void DrawHeadApparel(PawnRenderer renderer, Mesh mesh, Vector3 rootLoc, Vector3 headLoc, Vector3 headOffset, Rot4 bodyFacing, Quaternion quaternion, bool portrait, ref bool hideHair)
         {
             var apparelGraphics = renderer.graphics.apparelGraphics;
-            var headwearGraphics = apparelGraphics.Where(a => a.sourceApparel.def.apparel.LastLayer.GetModExtension<ApparelLayerExtension>()?.IsHeadwear ?? false).ToArray();
-            if (!headwearGraphics.Any())
+            List<ApparelGraphicRecord> headwearGraphics = new List<ApparelGraphicRecord>();
+
+            foreach(var apparel in apparelGraphics)
+            {
+                if(apparel.sourceApparel.def.apparel.LastLayer.GetModExtension<ApparelLayerExtension>()?.IsHeadwear ?? false) 
+                    headwearGraphics.Add(apparel);
+            }
+
+            if (headwearGraphics.Count == 0)
```

Comment:
Is the `foreach` here really better than keeping the original LINQ expression and replacing `.ToArray()` with `.ToList()` and using `.Count` like you have?

Reply:
Its a good question - Honestly not sure, I have the list because I was fiddling with trying to use a bool instead of the .Count/.Length call among a few other micro-optimisations. The difference I saw between these changes was (in my opinion) likely just noise from the Profiling/Stopwatch.

I'll regress back to a flat array and if the times remain consistent I'll push.